playcanvas/src/editor.js

```diff
@@ -0,0 +1,119 @@
+import { fetchCampaignAd, sendOnLoadMetric, sendOnClickMetric, AD_REFRESH_INTERVAL } from '../../utils/networking';
+import { openURL, visibilityCheck } from '../../utils/helpers';
+import { formats } from '../../utils/formats';
+
+/** @type {typeof import("playcanvas").ScriptType} */
+const ZestyBanner = pc.createScript('zesty-banner');
+
+ZestyBanner.attributes.add("adUnitId", { type: "string" });
+ZestyBanner.attributes.add("format", {
+    type: "number",
+    enum: [
+        { "Medium Rectangle": 1 },
+        { "Billboard": 2 },
+        { "Mobile Phone Interstitial": 3 },
+    ],
+    default: 1,
+});
+ZestyBanner.attributes.add("cameraEntity", { type: "entity" });
+
+const FORMATS = {
+    1: "medium-rectangle",
+    2: "billboard",
+    3: "mobile-phone-interstitial",
+}
+
+// initialize code called once per entity
+ZestyBanner.prototype.initialize = function() {
+    this.ctaUrl = "https://relay.zesty.xyz";
+    this.campaignId = "DefaultCampaign";
```

Comment:
Think default information like this could be kept at a level outside of the main directory in case there are changes to it in the future?

Reply:
Sure thing, I'll make these constants in networking.js and just export them from there